PWGHF/D2H/Tasks/taskLc.cxx

```diff
@@ -235,11 +267,28 @@ struct HfTaskLc {
     registry.add("MC/reconstructed/signal/hDecLenErrSig", "3-prong candidates (matched);decay length error (cm);entries", {HistType::kTH2F, {{100, 0., 1.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
     registry.add("MC/reconstructed/prompt/hDecLenErrSigPrompt", "3-prong candidates (matched, prompt);decay length error (cm);entries", {HistType::kTH2F, {{100, 0., 1.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
     registry.add("MC/reconstructed/nonprompt/hDecLenErrSigNonPrompt", "3-prong candidates (matched, non-prompt);decay length error (cm);entries", {HistType::kTH2F, {{100, 0., 1.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+
+    if (applyMl) {
+      const AxisSpec thnAxisMass{thnConfigAxisMass, "inv. mass (p K #pi) (GeV/#it{c}^{2})"};
+      const AxisSpec thnAxisPt{thnConfigAxisPt, "#it{p}_{T}(#Lambda_{c}^{+}) (GeV/#it{c})"};
+      const AxisSpec thnAxisPtProng0{thnConfigAxisPtProng0, "#it{p}_{T}(prong0) (GeV/#it{c})"};
+      const AxisSpec thnAxisPtProng1{thnConfigAxisPtProng1, "#it{p}_{T}(prong0) (GeV/#it{c})"};
+      const AxisSpec thnAxisPtProng2{thnConfigAxisPtProng2, "#it{p}_{T}(prong1) (GeV/#it{c})"};
+      const AxisSpec thnAxisMultiplicity{thnConfigAxisMultiplicity, "multiplicity"};
+      const AxisSpec thnAxisChi2PCA{thnConfigAxisChi2PCA, "Chi2PCA to sec. vertex (cm)"};
+      const AxisSpec thnAxisDecLength{thnConfigAxisDecLength, "decay length (cm)"};
+      const AxisSpec thnAxisDecLengthXY{thnConfigAxisDecLengthXY, "decay length xy (cm) "};
+      const AxisSpec thnAxisCPA{thnConfigAxisCPA, "cosine of pointing angle"};
+      const AxisSpec thnAxisCPAXY{thnConfigAxisCPAXY, "cosine of pointing angle xy"};
+      const AxisSpec thnAxisBdtScoreLcBkg{thnConfigAxisBdtScoreLcBkg, "BDT bkg score (Lc)"};
+      const AxisSpec thnAxisBdtScoreLcNonPrompt{thnConfigAxisBdtScoreLcNonPrompt, "BDT non-prompt score (Lc)"};
+      const AxisSpec thnAxisCanType{thnConfigAxisCanType, "candidates type"};
+      registry.add("hBdtScoreVsVars", "THn for Lambdac candidates", HistType::kTHnSparseF, {thnAxisMass, thnAxisPt, thnAxisPtProng0, thnAxisPtProng1, thnAxisPtProng2, thnAxisMultiplicity, thnAxisChi2PCA, thnAxisDecLength, thnAxisDecLengthXY, thnAxisCPA, thnAxisCPAXY, thnAxisBdtScoreLcBkg, thnAxisBdtScoreLcNonPrompt, thnAxisCanType});
```

Comment:
In O2 there is a maximum limit of axes that you can add to a multidimensional histogram (see https://github.com/AliceO2Group/AliceO2/blob/f09accf8cf210b047fb1fa0726a451510a72d8ad/Framework/Core/src/HistogramSpec.cxx#L82-L86), so if you put more than 10 axes it crashes.

Maybe you can foresee a version of the THnSparse with ML scores, and one with topological variables. Also consider that it will be very large (especially in Pb-Pb collisions, so having a small number of axes it could be a better idea).

Reply:
Thank you so much for reminding me about this! The reason I keep so many vars in the Thn is that can still let the analyser do the further rectangular cut locally. Then it's a good option to reduce the vars in the ML Thn, and keep more in the Std Thn.